core/base/src/TString.cxx

```diff
@@ -131,6 +131,11 @@ TString::TString(const char *cs, Ssiz_t n)
       Zero();
       return;
    }
+   for (int i = 0; i < strlen(cs)-1;i++) {
+      if (cs[i] == "\0") {
+         Error("TString::TString", "Warning. Input string contains non-terminating NULL character!");
+      }
+   }
```

Comment:
Hey, it looks like you are scanning for the `\0` character here in a for loop.
That's *exactly* what `strlen` does, so this doesn't make sense.

The check should read something like
```c++
if (strlen(cs) < n) {
  ...
```

Reply:
Okay, I have done this now.